TYPO3.Flow/Classes/TYPO3/Flow/Core/Bootstrap.php

```diff
@@ -526,6 +527,17 @@ protected function defineConstants()
         define('FLOW_PATH_DATA', FLOW_PATH_ROOT . 'Data/');
         define('FLOW_PATH_PACKAGES', FLOW_PATH_ROOT . 'Packages/');
 
+        if (!defined('FLOW_PATH_TEMPORARY')) {
+            $temporaryDirectoryPath = self::getEnvironmentConfigurationSetting('FLOW_PATH_TEMPORARY');
+            if ($temporaryDirectoryPath !== null) {
+                $temporaryDirectoryPath = rtrim($temporaryDirectoryPath, '/') . '/';
+            } else {
+                $temporaryDirectoryBasePath = self::getEnvironmentConfigurationSetting('FLOW_PATH_TEMPORARY_BASE') ?: FLOW_PATH_DATA . '/Temporary';
+                $temporaryDirectoryPath = Files::concatenatePaths(array($temporaryDirectoryBasePath, str_replace('/', '/SubContext', (string)$this->context))) . '/';
```

Comment:
I don't really understand the objection. The context was always part of the temporary path (see Environment:: createTemporaryDirectory()).
That's why I introduced a new env variable FLOW_PATH_TEMPORARY_BASE because the context sub folders always need to be added or you end up using the same temporary files for multiple contexts